### PR TITLE
feat(go-workflow,llm-tools): auto-generate tests for fixes during address-review

### LINKED ISSUE #75: feat(go-workflow): auto-generate tests for fixes during address-review
## Summary

When Claude Code fixes issues found during PR reviews (via `/address-review` or any review-related workflow), it should automatically write tests for those fixes. This also applies when review work is delegated to other LLMs via the `llm-tools` plugin (`/codex`, `/gemini`, `/ollama`, `/llm-compare`). Currently, fixes are applied without corresponding test coverage, which means the same issues could regress.

## Motivation

- Fixes without tests are incomplete — they solve the immediate problem but don't prevent regression
- Review comments often point to edge cases or incorrect behavior, which are ideal candidates for test cases
- Automating test generation alongside fixes ensures every review cycle improves the test suite
- This aligns with the existing TDD approach used in `/fix-issue` and `/add-feature`
- When delegating to external LLMs, test generation acts as a verification layer — ensuring the delegated fix is correct

## Scope

This applies to **all review-addressing and LLM-delegated workflows**, including:
- `/address-review` (go-workflow)
- `/codex` (llm-tools) — when Codex is delegated review fixes
- `/gemini` (llm-tools) — when Gemini is delegated review fixes
- `/ollama` (llm-tools) — when local models handle review fixes
- `/llm-compare` (llm-tools) — when comparing fix approaches across LLMs
- Any flow where Claude or a delegated LLM reads review comments, identifies issues, and applies fixes

## Expected Behavior

When a fix is applied in response to a review comment:
1. Identify what the fix changes (function, method, behavior)
2. **Determine if the change is testable** (see guidelines below)
3. If testable: write a test that would have caught the original issue
4. Verify the test fails without the fix and passes with it (red-green confirmation)
5. Include the test in the same commit as the fix

For LLM-delegated fixes specifically:
- The delegated LLM should be instructed to include tests alongside its fix
- If the delegated LLM does not produce tests, Claude should generate them after receiving the fix
- Tests serve as a correctness check on externally-generated code

## When to Write Tests vs. Skip

Not every fix is testable. Use judgment based on whether the change affects observable behavior.

**DO write tests for:**
- Bug fixes that change function/method output or behavior
- Logic changes (conditionals, error handling, edge cases)
- New or modified validation rules
- Data transformation or parsing changes
- API response changes
- Concurrency or race condition fixes
- Any change that alters what a function returns, how it mutates state, or what side effects it produces

**DO NOT write tests for:**
- Removing or adding comments
- Adding/changing log statements
- Formatting or whitespace changes
- Import reordering
- Variable/function renames (unless public API)
- Documentation updates
- Config file tweaks that don't affect runtime behavior
- Typo fixes in non-user-facing strings

**Rule of thumb:** If the change affects something a caller or user could observe — a return value, an error, a side effect, an HTTP response — it's testable and should get a test. If it's purely cosmetic or informational, skip it.

## Implementation Notes

- Leverage the existing `go-dev:test-gen` skill/patterns for Go test generation
- For table-driven tests, add a new test case to an existing table if one exists rather than creating a new test function
- The test should target the specific behavior the review comment flagged
- For `llm-tools` delegation, update the prompt templates to include "write tests for any testable fix" instructions

### PR DESCRIPTION
**Pull request by corylanou**
## Summary

- Add **Step 4.5** to `/address-review` that automatically generates tests for testable fixes after all review comments are addressed
- Expand Step 4a/4e tracking to classify each fix as testable or not, with affected function/package
- Update commit message template (Step 6) to mention test additions
- Add **Review Fix Detection** to `/codex`, `/gemini`, `/ollama`, and `/llm-compare` that appends test generation instructions to LLM prompts when review-fix context is detected
- Add **fallback** to each LLM command so Claude generates tests if the delegated LLM skips them

Fixes #75

## Test Plan

- [ ] Run `/address-review` on a PR with testable review comments (e.g., logic fix, error handling) — verify Step 4.5 generates tests
- [ ] Run `/address-review` on a PR with non-testable comments (formatting, comments) — verify Step 4.5 is skipped
- [ ] Run `/codex` with a review-fix prompt — verify test instructions are appended and fallback triggers if needed
- [ ] Run `/gemini` with a review-fix prompt — verify same behavior
- [ ] Run `/ollama` with a review-fix prompt — verify same behavior with local model note
- [ ] Run `/llm-compare` with a review-fix prompt — verify test instructions added to all LLM prompts and Test Coverage section appears in report
- [ ] Verify step numbering in address-review.md is intact (4 → 4.5 → 5 → ... → 12)
- [ ] Verify `./scripts/check-shared-sync.sh` passes

### CODE REVIEW

File: plugins/llm-tools/commands/gemini.md [hunk 1]
```diff
@@ -139,6 +139,23 @@ After execution completes:
 - Ask if they want a follow-up question or different perspective
 - Offer to try `/codex` or `/ollama` for comparison
 
+## Review Fix Detection
+
+Before running Gemini, detect if the prompt is addressing review feedback (e.g., contains phrases like "fix review comment", "address feedback", "fix the issue from review", or originates from an `/address-review` context). If a review-fix prompt is detected:
+
```
[the remark this answers is not in the document]
Comment (corylanou) [reply plugins/llm-tools/commands/gemini.md]: Fixed: Moved Review Fix Detection to step 4, before Run Gemini (now step 5). Updated step numbering accordingly.

**Comment by corylanou** [comment]
@codex review

File: plugins/llm-tools/commands/codex.md [hunk 2]
```diff
@@ -47,6 +47,65 @@ Run a task using OpenAI Codex CLI with the prompt: $ARGUMENTS
 Check if the prompt contains "review" (case-insensitive). If yes, route to **Review Flow**.
 Otherwise, continue with **Exec Flow**.
 
+## 2. Review Fix Detection (applies to both flows)
+
+Before running Codex in either flow, detect if the prompt is addressing review feedback (e.g., contains phrases like "fix review comment", "address feedback", "fix the issue from review", or the prompt originates from an `/address-review` context). If a review-fix prompt is detected:
```
[the remark this answers is not in the document]
Comment (corylanou) [reply plugins/llm-tools/commands/codex.md]: Fixed: Updated Step 1 routing to detect fix-oriented review prompts (containing "fix", "address", "resolve", "update" alongside "review") and route them to Exec Flow where files can be modified and test generation can run.
